src/api/createRequest.js

```diff
@@ -0,0 +1,148 @@
+import axios from 'axios'
```

Comment:
I've had a reasonable amount of success dropping axios entirely, and just using the native `fetch` calls. The main differences I noted between that approach and axios is:
1) Only actual network errors throw, so you'll need to check the status of the response yourself.
2) I needed to URI encode the parameters myself, which made me a bit sad.

Examples from where I did it:
```ruby
const plateRequest = async (plate_barcode) => {
  try {
    return await window.fetch(PLATE_PATH + encodeURIComponent(plate_barcode), {});
  } catch (error) {
    console.error(error);
    throw `Network Error: ${error.message}`;
  }
}
```

Then actually making that request:
```ruby
 // We've not loaded the plate yet, so go fetch it.
      try {
        const response = await plateRequest(plate_barcode);

        if (response.status === 200) {
          const plate = await extractPlateJson(response);
          commit("updatePlate", { ...plate, scanned: true });
          commit("updateScanPlate", { barcode: plate_barcode, id: plate.id });
          if (!plate.control) {
            dispatch("fetchBatches", { ids: plate.batches });
          }
        } else {
          const error_message = await extractErrors(response);
          throw error_message;
        }
      } catch (error) {
        // We catch and log most expected errors where they occur, and re-throw with
        // a user-friendly error message. This handles displaying that
        commit("updateScanPlate", { barcode: plate_barcode, errorMessage: error });
      }
```

Not suggesting we drop axios here, but this always seems to be the approach people recommend whenever anyone runs into issues with axios.

Reply:
Happy to leave it for now but as everything is in one place it would be easy to update if we ever run into problems.